mysqlnd_azure.c

```diff
@@ -340,23 +365,38 @@ MYSQLND_METHOD(mysqlnd_azure_data, connect)(MYSQLND_CONN_DATA ** pconn,
         unsigned int ui_redirect_port = 0;
         zend_bool serverSupportRedirect = get_redirect_info(conn, redirect_host, redirect_user, &ui_redirect_port);
         if (!serverSupportRedirect) {
+          if (MYSQLND_AZURE_G(logLevel) > 0) {
+            AZURE_LOG("ERROR", "Failed to get REDIRECTION information from OK Packet, please check whether your MySQL server support redirect.");
+          }
             DBG_ENTER("[redirect]: Server does not support redirection.");
             if(MYSQLND_AZURE_G(enableRedirect) == REDIRECT_ON) {
                 //REDIRECT_ON, if redirection is not supported, abort the original connection and return error
                 conn->m->send_close(conn);
+                if (MYSQLND_AZURE_G(logLevel) > 0) {
+                  AZURE_LOG("ERROR", "mysqlnd_azure.Redirection: ON, MySQL server not support REDERECTION, connection failed");
+                }
                 SET_CLIENT_ERROR(conn->error_info, MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO, UNKNOWN_SQLSTATE, "Connection aborted because redirection is not enabled on the MySQL server or the network package doesn't meet meet redirection protocol.");
                 goto err;
             } else {
+              if (MYSQLND_AZURE_G(logLevel) > 1) {
+                AZURE_LOG("INFO ", "mysqlnd_zaure.Redirection: PREFERRED, MySQL server not support REDIRECTION, conn falls back to classical one.");
+              }
                 //REDIRECT_PREFERRED, do nothing else for redirection, just use the previous connection
                 goto after_conn;
             }
         }
 
         //Get here means serverSupportRedirect
 
+        if (MYSQLND_AZURE_G(logLevel) > 2) {
+          AZURE_LOG("DEBUG", "Successfully get redirection information, try to connect with redirection connection");
+        }
         //Already use redirected connection, or the connection string is a redirected one
         if (strcmp(redirect_host, hostname.s) == 0 && strcmp(redirect_user, username.s) == 0 && ui_redirect_port == port) {
             DBG_ENTER("[redirect]: Is using redirection, or redirection info are equal to origin, no need to redirect");
+            if (MYSQLND_AZURE_G(logLevel) > 2) {
```

Comment:
Here connection is already established. So maybe "do nothing for redirection." instead of "do nothing and just connect it."?